src/sage/knots/link.py

```diff
@@ -2362,6 +2403,48 @@ def regions(self):
             regions.append(region)
         return regions
 
+    def remove_loops(self):
+        r"""
+        Return an ambient isotopic link in which all loops are removed.
+
+        EXAMPLES::
+
+          sage: b = BraidGroup(4)((3, 2, -1, -1))
+          sage: L = Link(b)
+          sage: L.remove_loops()
+          Link with 2 components represented by 2 crossings
+          sage: K4 = Link([[1, 7, 2, 6], [3, 1, 4, 8], [5, 5, 6, 4], [7, 3, 8, 2]])
+          sage: K3 = K4.remove_loops()
+          sage: K3.pd_code()
+          [[1, 7, 2, 4], [3, 1, 4, 8], [7, 3, 8, 2]]
+          sage: U = Link([[1, 2, 2, 1]])
+          sage: U.remove_loops()
+          Link with 1 component represented by 0 crossings
+        """
+        pd = self.pd_code()
+        from copy import copy
+        new_pd = [copy(cr) for cr in pd if len(set(cr)) == 4]
+        if not new_pd:
+            # trivial knot
+            return type(self)([])
+        elif pd == new_pd:
+            # no loops detected
+            return self
+        from sage.sets.set import Set
+        new_edges = Set(flatten(new_pd))
+        for cr in Set(pd).difference(Set(new_pd)):
+            # cr is a loop crossing
+            rem = list(Set(cr).intersection(new_edges))
```

Comment:
Wouldn't it also be better to simply split the PD code into the two parts when constructing `new_pd` code? That way you could avoid having to use Sage `Set`'s and only need to pass over the PD code once.

Reply:
I will try to do this later on!

Reply:
Done in https://github.com/sagemath/sage/pull/36884/commits/e0b5e55a624c30a4f8d9582b8e9601ac7c93f749.